unikernel.ml

```diff
@@ -779,6 +779,128 @@ struct
+    match User_model.find_user_by_key uuid users with
+    | Some u ->
+        let user_policy =
+          match Albatross.policy albatross ~domain:u.name with
+          | Ok p -> (
+              match p with Some p -> p | None -> Albatross.empty_policy)
+          | Error _ -> Albatross.empty_policy
+        in
+        let policy_avalaible = Albatross.policy_resource_used () albatross in
+        Lwt.return
+          (reply reqd ~content_type:"text/html"
+             (Dashboard.dashboard_layout user
+                ~page_title:(String.capitalize_ascii u.name ^ " | Mollymawk")
+                ~content:
+                  (Update_policy.update_policy_layout u ~user_policy
+                     ~root_policy:policy_avalaible)
+                ~icon:"/images/robur.png" ())
+             `OK)
+    | None ->
+        let status =
+          {
+            Utils.Status.code = 404;
+            title = "Error";
+            data = "Couldn't find account with uuid: " ^ uuid;
+            success = false;
+          }
+        in
+        Lwt.return
+          (reply reqd ~content_type:"text/html"
+             (Guest_layout.guest_layout ~page_title:"404 | Mollymawk"
+                ~content:(Error_page.error_layout status)
+                ~icon:"/images/robur.png" ())
+             `Not_found)
+
+  let update_policy store albatross reqd _user =
+    decode_request_body reqd >>= fun data ->
+    let json =
+      try Ok (Yojson.Basic.from_string data)
+      with Yojson.Json_error s -> Error (`Msg s)
+    in
+    match json with
+    | Error (`Msg err) ->
+        Logs.warn (fun m -> m "Failed to parse JSON: %s" err);
+        http_response reqd ~title:"Error" ~data:(String.escaped err)
+          `Bad_request
+    | Ok json -> (
+        match json with
+        | `Assoc xs -> (
+            match
+              Utils.Json.
+                ( get "vms" xs,
+                  get "memory" xs,
+                  get "block" xs,
+                  get "cpuids" xs,
+                  get "bridges" xs,
+                  get "user_uuid" xs )
+            with
+            | ( Some (`Int vms),
+                Some (`Int memory),
+                Some (`Int block),
+                Some (`String cpuids),
+                Some (`String bridges),
+                Some (`String user_uuid) ) -> (
+                let users =
+                  User_model.create_user_uuid_map (snd store).Storage.users
+                in
+                match User_model.find_user_by_key user_uuid users with
+                | Some u -> (
+                    let policy_data =
+                      {
+                        Vmm_core.Policy.vms;
+                        memory;
+                        block = Some block;
+                        cpuids =
+                          (if cpuids = "" then Vmm_core.IS.empty
+                           else
+                             Vmm_core.IS.of_list
+                               (List.map int_of_string
```

Comment:
int_of_string may raise an exception, and this is user input -- we need to handle this properly (use a try .. catch or the int_of_string_opt)